lib/streamlit/util.py

```diff
@@ -258,12 +258,14 @@ def open_browser(url):
 
     system = platform.system()
 
-    if system == "Windows":
+    if (system == "Windows"
+        or system == "Linux" and not _is_executable_in_path("xdg-open")):
```

Comment:
Please merge `develop`. I just did some surgery on `util.py`.

Also added two constants you can use here: `env_util.IS_WINDOWS` and `env_util.IS_LINUX_OR_BSD`